### PR TITLE
feat(meta): Add `creating_status` field for stream jobs

### PR DESCRIPTION
**Pull request by kwannoel**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Support identifying `CREATING` / `CREATED` tables.

Part of #8051 

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/frontend/src/catalog/table_catalog.rs [hunk 1]
```diff
@@ -401,6 +403,7 @@ impl TableCatalog {
             initialized_at_epoch: self.initialized_at_epoch.map(|epoch| epoch.0),
             created_at_epoch: self.created_at_epoch.map(|epoch| epoch.0),
             cleaned_by_watermark: self.cleaned_by_watermark,
+            create_status: PbTableStatus::Creating.into(),
```
Comment (yezizp2012) [inline src/frontend/src/catalog/table_catalog.rs:406]: Note that `to_prost` is not only used for streaming job creation, but also used for `alter_table_column`, which will replace the whole catalog for target table. Making the status default to `Creating` is acceptable to me, but you need to set it to `Create` in meta side not only for creation but also for table altering. You can do it in this PR or in later one.

File: proto/catalog.proto [hunk 2]
```diff
@@ -32,6 +32,13 @@ enum SchemaRegistryNameStrategy {
   TOPIC_RECORD_NAME_STRATEGY = 2;
 }
 
+enum StreamJobStatus {
+  // Prefixed by `STREAM_JOB_STATUS` due to protobuf namespacing rules.
+  STREAM_JOB_STATUS_UNSPECIFIED = 0;
```
Comment (yezizp2012) [inline proto/catalog.proto:37]:
```suggestion
  UNSPECIFIED = 0;
```

**Review by yezizp2012 — APPROVED** [review]
Rest LGTM!

File: proto/catalog.proto (hunk 2 above)
Comment (kwannoel) [reply proto/catalog.proto:37]: Protobuf namespacing rules follow c++. `SinkType` already has `UNSPECIFIED` as a variant, so it will conflict.

Comment (kwannoel) [reply proto/catalog.proto:37]:
```
  Failed to compile grpc!: Custom { kind: Other, error: "protoc failed: catalog.proto:95:3: \"UNSPECIFIED\" is already defined in \"catalog\".\ncatalog.proto:95:3: Note that enum values use C++ scoping rules, meaning that enum values are siblings of their type, not children of it.  Therefore, \"UNSPECIFIED\" must be unique within \"catalog\", not just within \"SinkType\".\nbatch_plan.proto:5:1: Import \"catalog.proto\" was not found or had errors.\nbatch_plan.proto:66:3: \"catalog.StreamSourceInfo\" is not defined.\n" }
```

File: src/frontend/src/catalog/table_catalog.rs (hunk 1 above)
Comment (kwannoel) [reply src/frontend/src/catalog/table_catalog.rs]: Thanks for mentioning it. Will add it here.

File: proto/catalog.proto (hunk 2 above)
Comment (yezizp2012) [reply proto/catalog.proto:37]: Oh, I see. 👍 Just forget it and get this PR merged first.

File: src/frontend/src/catalog/table_catalog.rs (hunk 1 above)
Comment (kwannoel) [reply src/frontend/src/catalog/table_catalog.rs]: https://github.com/risingwavelabs/risingwave/pull/12342

**Pull request merged**